Polygon_mesh_processing/include/CGAL/Envelope.h

```diff
@@ -0,0 +1,1901 @@
+//
+// This file incorporates work covered by the following copyright and permission notice:
+//
+//     MIT License
+//
+//     Copyright (c) 2019 Bolun Wang, Teseo Schneider, Yixin Hu, Marco Attene, and Daniele Panozzo
+//
+//     Permission is hereby granted, free of charge, to any person obtaining a copy
+//     of this software and associated documentation files (the "Software"), to deal
+//     in the Software without restriction, including without limitation the rights
+//     to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+//     copies of the Software, and to permit persons to whom the Software is
+//     furnished to do so, subject to the following conditions:
+//
+//     The above copyright notice and this permission notice shall be included in all
+//     copies or substantial portions of the Software.
+//
+//     THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//     IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//     FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//     AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//     LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//     OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+//     SOFTWARE.
+//
+//
+//
+// @article{Wang:2020:FE,
+//    title={Exact and Efficient Polyhedral Envelope Containment Check},
+//    author={Bolun Wang and Teseo Schneider and Yixin Hu and Marco Attene and Daniele Panozzo},
+//    journal = {ACM Trans. Graph.},
+//     volume = {39},
+//     number = {4},
+//     month = jul,
+//     year = {2020},
+//     publisher = {ACM}
+// }
+//
+// The code below uses the version of
+// https://github.com/wangbolun300/fast-envelope  avaiable on 7th of October 2020
+//
+// The code below only use the high level algorithms of checking that a query
+// is covered by a set of prisms, where each prism is an offset for an input triangle.
+// That is, we do not use indirect predicates
+
+#ifndef CGAL_POLYGON_MESH_PROCESSING_ENVELOPE_H
+#define CGAL_POLYGON_MESH_PROCESSING_ENVELOPE_H
+
+#include <CGAL/license/Polygon_mesh_processing/Envelope.h>
+
+#include <CGAL/disable_warnings.h>
+
+#include <CGAL/Exact_predicates_exact_constructions_kernel.h>
+
+#include <CGAL/AABB_tree.h>
+#include <CGAL/AABB_traits.h>
+#include <CGAL/AABB_primitive.h>
+
+#ifdef CGAL_ENVELOPE_DEBUG
+// This is for computing the surface mesh of a prism
+#include <CGAL/Convex_hull_3/dual/halfspace_intersection_3.h>
+#include <CGAL/Surface_mesh.h>
+#include <CGAL/boost/graph/copy_face_graph.h>
+#endif
+
+#include <CGAL/boost/graph/named_params_helper.h>
+
+#include <boost/iterator/counting_iterator.hpp>
+
+#include <string>
+#include <fstream>
+
+
+
+namespace CGAL {
+
+/**
+ * \ingroup PkgPolygonMeshProcessingRef
+ * This class provides a test whether a query point, segment, or triangle
```

Comment:
I'm not sure the use of "whether" is correct here.

Reply:
polyhedral envelope should be defined or a link to the user manual should be added.

Reply:
What about   "This class can be used to check if a query point, segment, or triangle is inside or outside a polyhedral envelope of a triangle mesh."